src/flows/pipes/QueryBuilder/CardList.tsx

```diff
@@ -185,7 +190,7 @@ const Card: FC<Props> = ({idx}) => {
   } else {
     _values = (
       <SelectorList
-        items={card.values.results}
+        items={joinAndDedupeArrays(card.values.results, card.values.selected)}
```

Comment:
thoughts on moving this up and wrapping it in a `useMemo`? I'm thinking this component probably gets re-rendered a lot and if there are hundreds of buckets this might be an expensive operation maybe?